storage_service/locations/models/dataverse.py

```diff
@@ -77,31 +147,61 @@ def browse(self, path):
                 raise StorageException(
-                    _('Unable to fetch datasets from %(url)s with query %(path)s') %
-                    {'url': url, 'path': path})
+                    _('Unable to fetch datasets from %(url)s')
+                    % {'url': url})
             try:
                 data = response.json()['data']
+                items = data['items']
             except json.JSONDecodeError:
                 LOGGER.error('Could not parse JSON from response to %s', url)
                 raise StorageException(
-                    _('Unable parse JSON from response to %(url)s with query %(path)s') %
-                    {'url': url, 'path': path})
-
-            entries += [str(x['entity_id']) for x in data['items']]
-
-            properties.update({
-                str(x['entity_id']): {'verbose name': x['name']}
-                for x in data['items']
-            })
-
+                    _('Unable to parse JSON from response to %(url)s')
+                    % {'url': url})
+            entries += [str(ds['entity_id']) for ds in items]
+            properties.update(
+                {str(ds['entity_id']): {'verbose name': ds['name']}
+                 for ds in items})
             if params['start'] + data['count_in_response'] < data['total_count']:
                 params['start'] += data['count_in_response']
             else:
                 break
+        return {
+            'directories': entries,
+            'entries': entries,
+            'properties': properties,
+        }
 
-        directories = entries
+    def _browse_dataset(self, dataset_identifier):
+        """Return all files in the dataset with ``entity_id``
+        ``dataset_identifier`` (conforming to ``browse`` protocol).
+        """
+        files_in_dataset_path = (
+            '/api/v1/datasets/{dataset_identifier}/versions/:latest'.format(
+                dataset_identifier=dataset_identifier))
+        url = 'https://{}{}'.format(self.host, files_in_dataset_path)
+        params = {'key': self.api_key}
+        LOGGER.debug('URL: %s, params: %s', url, params)
+        response = requests.get(url, params=params)
+        LOGGER.debug('Response: %s', response)
+        if response.status_code != 200:
+            LOGGER.warning('%s: Response: %s', response, response.text)
+            raise StorageException(
+                _('Unable to fetch datasets from %(url)s')
+                % {'url': url})
+        try:
+            data = response.json()['data']
+            files = data['files']
+        except json.JSONDecodeError:
+            LOGGER.error('Could not parse JSON from response to %s', url)
+            raise StorageException(
+                _('Unable to parse JSON from response to %(url)s')
+                % {'url': url})
+        entries = [f['dataFile']['filename'] for f in files]
+        properties = {
+            f['dataFile']['filename']: {'size': f['dataFile']['filesize']}
+            for f in files}
```

Comment:
Small thing, but if the `properties` dict comp were performed first, then the entries could be derived from it via `entries = list(properties.keys())`. Same comment applies to the similar construction in _browse_dataverse.

Reply:
This is a nice shortcut. Good thinking!

Reply:
The `entries = []` line in `_browse_dataverse` is no longer necessary. It could be removed and `entries += list(properties.keys())` could become `entries = list(properties.keys())`.